script/locking_runner

```diff
@@ -5,6 +5,13 @@
 
 SEMAPHORE=$1
 LOCKFILE=tmp/pids/$SEMAPHORE
+
+if ( [ -e $LOCKFILE ] && ! kill -0 $(< $LOCKFILE) ) 2> /dev/null;
```

Comment:
`kill -0` will fail if the user doesn't have permission to send signal to the process, which could happen in some corner cases. Did you consider `ps -p`, since it doesn't have this problem?